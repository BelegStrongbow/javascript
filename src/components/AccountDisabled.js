import React from "react";
import pageNotFoundImage from "../images/PageNotFound.svg";
import { defineMessages, injectIntl, intlShape, FormattedMessage } from "react-intl";
import speak from "a11y-speak";
import styled from "styled-components";
import logo from "../images/logoColor.svg";
import { Heading } from "./Headings";

const LogoImage = styled.img`
	width: ${ props => props.size };
	display: block;
	margin: 30px auto 25px;
`;

const PageContainer = styled.div`
	margin-top: 10vh;
	text-align: center;
`;

const AccountDisabledImage = styled.img`
	width: 384px;
	max-width: 100%;
	height: auto;
	margin-top: 2em;
`;

const messages = defineMessages( {
	accountDisabled: {
		id: "account.disabled.shortMessage",
		defaultMessage: "Account disabled",
	},
	supportTeam: {
		id: "account.disabled.supportTeam",
		defaultMessage: "support team",
	},
} );

/**
 * A function that returns the Account disabled component.
 *
 * @returns {ReactElement} The component that contains the Account disabled page.
 */
class AccountDisabled extends React.Component {
	componentDidMount() {
		// Announce navigation to assistive technologies.
		let message = this.props.intl.formatMessage( messages.accountDisabled );
		speak( message );
	}

	render() {
		let paragraphs = [
			<FormattedMessage id="account.disabled.possibleAction" defaultMessage={ "If you would like to re-enable your account, please contact our { support }." }
							  values={ { support: <a href="mailto:support@yoast.com">support team</a> } } />,
			<FormattedMessage id="account.disabled.visitYoast" defaultMessage={ "Visit { yoast }." }
							  values={ { yoast: <a href="https://yoast.com">Yoast.com</a> } } />,
		];
		return (


		<PageContainer>
			<LogoImage src={ logo } size="200px" />
			<Heading><FormattedMessage id="account.disabled.header" defaultMessage={ "Account disabled" } /></Heading>
			{ paragraphs.map( function( paragraph ) {
				return <p key={ paragraph.props.id }>{ paragraph }</p>;
			} ) }
			<AccountDisabledImage src={ pageNotFoundImage } alt="" />
		</PageContainer>
		);
	}
}

export default injectIntl( AccountDisabled );

AccountDisabled.propTypes = {
	intl: intlShape.isRequired,
};
