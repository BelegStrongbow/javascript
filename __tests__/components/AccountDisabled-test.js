import React from 'react';
import AccountDisabled from '../../src/components/AccountDisabled';
import { createComponentWithIntl } from "../../utils";

test('the account disabled component matches the snapshot', () => {
	const component = createComponentWithIntl(
		<AccountDisabled />
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
