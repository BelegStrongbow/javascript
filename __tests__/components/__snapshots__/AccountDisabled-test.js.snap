// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the account disabled component matches the snapshot 1`] = `
.c2 {
  font-weight: 700;
  font-size: 2em;
}

.c1 {
  width: 200px;
  display: block;
  margin: 30px auto 25px;
}

.c0 {
  margin-top: 10vh;
  text-align: center;
}

.c3 {
  width: 384px;
  max-width: 100%;
  height: auto;
  margin-top: 2em;
}

<div
  className="c0"
>
  <img
    className="c1"
    size="200px"
    src="logoColor.svg"
  />
  <h1
    className="c2"
  >
    <span>
      Account disabled
    </span>
  </h1>
  <p>
    <span>
      If you would like to re-enable your account, please contact our 
      <a
        href="mailto:support@yoast.com"
      >
        support team
      </a>
      .
    </span>
  </p>
  <p>
    <span>
      Visit 
      <a
        href="https://yoast.com"
      >
        Yoast.com
      </a>
      .
    </span>
  </p>
  <img
    alt=""
    className="c3"
    src="PageNotFound.svg"
  />
</div>
`;
